packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -584,10 +599,44 @@ export class CfnInclude extends core.CfnElement {
     return cfnCondition;
   }
 
+  private getPlaceholderID(): string {
+    let id = 'Placeholder';
+    for (let i = 0; i < 26; i++) {
+      id = id + randomAlphaNumeric();
+    }
+    return id;
+
+    function randomAlphaNumeric(): string {
+      if (Math.random() < 0.5) {
+        return String.fromCharCode('a'.charCodeAt(0) + Math.floor(Math.random() * 26));
+      } else {
+        return Math.floor(Math.random() * 10).toString();
+      }
+    }
+  }
+
   private getOrCreateResource(logicalId: string, cycleChain: string[] = []): core.CfnResource {
     cycleChain = cycleChain.concat([logicalId]);
     if (cycleChain.length !== new Set(cycleChain).size) {
-      throw new Error(`Found a cycle between resources in the template: ${cycleChain.join(' depends on ')}`);
+      if (!this.allowCyclicalReferences) {
+        throw new Error(`Found a cycle between resources in the template: ${cycleChain.join(' depends on ')}`);
+      }
+      let placeholderResourceAttributes: any = this.template.Resources[logicalId];
+      let placeholderId: string = this.getPlaceholderID();
+      //handle multiple cycles
+      if (this.logicalIdToPlaceholderMap.get(logicalId)) {
+        this.logicalIdToPlaceholderMap.get(logicalId)!.add(placeholderId);
```

Comment:
I'm feeling very dumb, but why have multiple of these? Couldn't they all use the same placeholder object?

Reply:
Multiple cycles with the same destination can result in two placeholders existing for the same logicalId :) see the multi-cycle-multi-dest unit I added.

Reply:
Actually, now I get it. Will test.